src/reflect/scala/reflect/internal/BaseTypeSeqs.scala

```diff
@@ -171,7 +171,7 @@ trait BaseTypeSeqs {
   /** A marker object for a base type sequence that's no yet computed.
    *  used to catch inheritance cycles
    */
-  val undetBaseTypeSeq: BaseTypeSeq = newBaseTypeSeq(List(), Array())
+  lazy val undetBaseTypeSeq: BaseTypeSeq = newBaseTypeSeq(List(), Array())
```

Comment:
This is necessary because it is invoked in the initializer otherwise, calling `newBaseTypeSeq` which depends on `statistics`. When this class is initialized, `statistics` is null, so making it `lazy` sidesteps the `NullPointerException` that we would receive otherwise.